tests/org.eclipse.n4js.flowgraphs.tests/model/reactFiles/react.n4jsd

```diff
@@ -0,0 +1,283 @@
+    props: Object =,
+    ...children: union{string, Element, Array<Element>}): Element;
+
+/**
+ * Verifies the object is an Element.
+ */
+export external public function isValidElement(object: Object): boolean;
+
+/**
+ * Render a Element to its initial HTML. This should only be used on the server.
+ * React will return an HTML string.
+ * You can use this method to generate HTML on the server and send the markup down on the initial
+ * request for faster page loads and to allow search engines to crawl your pages for SEO purposes.
+ * If you call R.render() on a node that already has this server-rendered markup,
+ * React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.
+ */
+export external public function renderToString(Element: Element): string;
+
+/**
+ * Similar to renderToString, except this doesn't create extra DOM attributes such as data-react-id,
+ * that React uses internally.
+ * This is useful if you want to use React as a simple static page generator, as stripping away the
+ * extra attributes can save lots of bytes.
+ */
+export external public function renderToStaticMarkup(Element: Element): string;
+
+export external public interface ~PropConstraint {
+    public get isRequired(): PropConstraint;
+}
+export external public interface ~PropType extends PropConstraint {}
+
+/**
+ * R.PropTypes includes types that can be used with a component's propTypes object to validate
+ * props being passed to your components.
+ *
+ * @see http://facebook.github.io/react/docs/reusable-components.html
+ */
+export external public class ~PropTypes {
+    // You can declare that a prop is a specific JS primitive. By default, these
+    // are all optional.
+    public const array: PropType;
+    public const bool: PropType;
+    public const func: PropType;
+    public const number: PropType;
+    public const object: PropType;
+    public const string: PropType;
+
+    // Anything that can be rendered: numbers, strings, Elements or an array
+    // containing these types.
+    public const node: PropType;
+
+    // A React Element.
+    public const Element: PropType;
+
+    // You can also declare that a prop is an instance of a class. This uses
+    // JS's instanceof operator.
+    public static instanceOf(comp: Object): PropConstraint;
+
+    // You can ensure that your prop is limited to specific values by treating
+    // it as an enum.
+    public static oneOf(...values: any): PropConstraint;
+
+    // An object that could be one of many types
+    public static oneOfType(...constraints: PropConstraint): PropConstraint;
+
+    // An array of a certain type
+    public static arrayOf(elemType: PropType): PropConstraint;
+
+    // An object with property values of a certain type
+    public static objectOf(valueType: PropType): PropConstraint;
+
+    // An object taking on a particular shape
+    public static shape(Object): PropConstraint;
+}
+
+/* -*- mode: typescript;-*- */
```

Comment:
we do not need this line anymore. it syntax should be highlighted based on our git config.